pandas/core/frame.py

```diff
@@ -8559,6 +8580,54 @@ def blk_func(values):
                 out[:] = coerce_to_dtypes(out.values, df.dtypes)
             return out
 
+        def array_func(values):
+            if isinstance(values, ExtensionArray):
+                return values._reduce(name, skipna=skipna, **kwds)
+            else:
+                return op(values, skipna=skipna, **kwds)
```

Comment:
why not pass axis here?  that looks like the only difference between this and blk_func

Reply:
Yes, blocks are 2D and thus the `blk_func` needs an axis, while this is working on the "column arrays", which are always 1D

Reply:
would passing axis=0 be harmful?  if not, can we de-duplicate these?

Reply:
> would passing axis=0 be harmful? if not, can we de-duplicate these?

For blocks, the axis is flipped, since the array is oriented differently. So if we share it, it would need some logic for which value of `axis` to pass. At which point the benefit of sharing might be lost.